cowbird/permissions_synchronizer.py

```diff
@@ -184,6 +187,13 @@ def _generate_regex_from_segments(res_segments: List[ConfigSegment]) -> Tuple[st
         for segment in res_segments:
             matched_groups = re.match(NAMED_TOKEN_REGEX, segment["name"])
             if matched_groups:
+                if segment.get("regex") is not None:
+                    # if a regex is passed, override the current regex and return
+                    regex = segment.get("regex")
+                    res_regex = (
+                        rf"{regex}"
+                    )
+                    return res_regex, -1
```

Comment:
It would ignore the following segment in the config, the regex in the source target should be used to extract what is needed in the path. (Should be used only in the last segment of the config for the source keys). I don't see the case where there would need to be two different regex in different segments of the same source key, since it would be the same as to have the regex combined and used in the last segment. For the target in the config , you can have multiple regex to match each segments differently (this is needed because `_create_res_data` use the regex to create the right type for the segments that are matched )

Reply:
> It would ignore the following segment in the config

Doesn't it also ignores the previous segments in the config too, since if a `regex` field is found it returns only that regex value?

Reply:
Ah yes, it will only uses that regex to extract what needs to be match.

Reply:
I wonder, if we have a `regex` field that overrides all the segments of a resource from the config, wouldn't it make more sense to have the `regex` field be one step higher in the config hierarchy instead of being found in one of the name/type pairs?
For example, either the resource_key uses a regex ...
```
services:
  <service_name>:
    <resource_key>:
      regex: <regex>
```
or it uses the current name/type settings?
```
services:
  <service_name>:
    <resource_key>:
      - name: <name_1>
        type: <type_1>
      - name: <name_2>
        type: <type_2>
      ...
```
I think the structure in my example would need more refinement, but the point is mostly that using a regex doesn't seem to use any of the name/type data, so the regex field might not need to be inserted in the name/type structure, but could be outside of it?

Reply:
That would be true for the source key, but for the target key the `regex` needs the type in the same segment to be able to recreate the right structure and it also allow to have multiple different regex in the target.